api/src/test/java/org/openmrs/api/ProgramWorkflowServiceTest.java

```diff
@@ -126,6 +126,217 @@ public void savePatientProgram_shouldUpdatePatientProgram() {
 		assertEquals(today, ptProg.getDateCompleted());
 		
 	}
+
+	/**
+	 * Tests if the savePatientProgram(PatientProgram) sets the EndDate of recent state of each workflow
+	 * on calling the setDateCompleted(Date date) method.
+	 *
+	 * @see ProgramWorkflowService#savePatientProgram(PatientProgram)
+	 */
+	@Test
+	public void savePatientProgram_onSetDateCompleted_shouldSetEndDateOfAllRecentStates() throws Exception {
```

Comment:
Would be better if this were named something like

`savePatientProgram_shouldSetEndDateOfAllRecentStatesWhenCompletingTheProgram`

I.e. have one underscore